src/lib/repositories/impl/order_repository_impl.py

```diff
@@ -30,5 +30,28 @@ def update_by_id(self, order_id, order):
     def get_orders_to_process(self):
         orders = self.get_all()
 
-        orders_to_process = filter(lambda order: order.status == OrderStatus.NEW_ORDER, orders)
+        orders_to_process = filter(
+            lambda order: order.status == OrderStatus.NEW_ORDER, orders
+        )
         return list(orders_to_process)
+
+    def get_chefs_with_assigned_orders(self, chefs_id):
+
+        orders = self.get_all()
+        assigned_chef_map = {}
+
+        for chef_id in chefs_id:
```

Comment:
I think you can use python reduce, could you investigate?

Reply:
done.